irohad/ametsuchi/impl/postgres_wsv_command.cpp

```diff
@@ -0,0 +1,227 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ametsuchi/impl/postgres_wsv_command.hpp"
+#include <iostream>
+
+namespace iroha {
+  namespace ametsuchi {
+
+    PostgresWsvCommand::PostgresWsvCommand(pqxx::nontransaction &transaction)
+        : transaction_(transaction) {}
+
+    bool PostgresWsvCommand::insertAccount(const model::Account &account) {
+      pqxx::binarystring master_key(account.master_key.data(),
+                                    account.master_key.size());
+      std::stringstream permissions;
+      permissions << account.permissions.add_signatory
+                  << account.permissions.can_transfer
+                  << account.permissions.create_accounts
+                  << account.permissions.create_assets
+                  << account.permissions.create_domains
+                  << account.permissions.issue_assets
+                  << account.permissions.read_all_accounts
+                  << account.permissions.remove_signatory
+                  << account.permissions.set_permissions
+                  << account.permissions.set_quorum;
+      try {
+        transaction_.exec(
+            "INSERT INTO account(\n"
+            "            account_id, domain_id, master_key, quorum, status, "
+            "transaction_count, \n"
+            "            permissions)\n"
+            "    VALUES (" +
+            transaction_.quote(account.account_id) + ", " +
+            transaction_.quote(account.domain_name) + ", " +
+            transaction_.quote(master_key) + ", " +
+            transaction_.quote(account.quorum) + ", " +
+            /*account.status*/ transaction_.quote(0) + ", " +
+            /*account.transaction_count*/ transaction_.quote(0) +
+            ", \n"
+            "            " +
+            transaction_.quote(permissions.str()) + ");");
+      } catch (const std::exception &e) {
+        std::cerr << e.what() << std::endl;
+        return false;
+      }
+      return true;
+    }
+
+    bool PostgresWsvCommand::insertAsset(const model::Asset &asset) {
+      uint32_t precision = asset.precision;
```

Comment:
For some reason we decided `precision` to be `uint8_t`, which is not supported in PostgreSQL, so it is casted here to `uint32_t`